packages/algorand-js/src/types.ts

```diff
@@ -80,7 +66,9 @@ export interface AccountsMap {
 
 export interface State {
   accounts: Map<string, StoreAccount>
+  accountAssets: Map<string, Map<number, AccountAssetInfo>>
   globalApps: Map<number, SSCParams>
+  globalAssets: Map<number, AssetParams>
```

Comment:
Why don't we keep assets in `StoreAccount` object?
I would see it as this:
```
type StoreAccount {
  ...
  assetHoldings: AssetHolding // renamed from AccountAssetInfo
}
```
`globalAssets` should be renamed to `assetDefs`  and `AssetParams` to `AssetDef`.
We can keep it directly State, or move it to `StoreAccount` as well. What do you think?

Reply:
I don't think we can move globalAssets to StoreAccount because `storeAccount` contains information linked to each account.
But we need have to information about assets globally, so that we can return queries about AssetParams

Reply:
assetHoldings is already in StoreAccount in `assets` object.
globalAssets is for Assets Parameters(Total, decimals, clawback etc..)

Reply:
AFAIK, in Algorand, all assets definitions are also stored in an Account. That's why I was thinking to put it there as well. But if it's not needed, or easier to do it other way around then I'm fine to keep `assetDefs` directly in `Store`.

Reply:
> assetHoldings is already in StoreAccount in assets object.

Then why did you add `accountAssets`?

Reply:
because I think this is faster. For ex: If we want to get user asset balance, One way is to traverse the whole assets array and stop when we find required `asset-id` for every query.
I think this will be slow, that's why `accountAssets` traverse accounts during initialization and creates a map<asset-id, asset-holdings>. when querying we get the values directly from map

> AFAIK, in Algorand, all assets definitions are also stored in an Account. That's why I was thinking to put it there as well. But if it's not needed, or easier to do it other way around then I'm fine to keep `assetDefs` directly in `Store`.

yes, they are stored in ["created-assets"] in every account, but for a query we  will have to traverse all the accounts to get a specific asset parameter. that's why i though global map for assets is better.

Reply:
If you want i can move assetDefs to store, and move the logic in storeAccount, (find value by traversing)

Reply:
OK, let's keep `assetDefs` outside of StoreAccount.

Reply:
>  If we want to get user asset balance, One way is to traverse the whole assets array and stop when we find required asset-id for every query.

But we still need to go through a user, right? user_account_address => asset_id => user_asset_holding

Reply:
accountAssets handles both, accountAssets contains <address, assets> and assets is <assetId, assetHoldings>